packages/@react-aria/focus/stories/FocusScope.stories.tsx

```diff
@@ -116,3 +123,55 @@ KeyboardNavigationNoContain.args = {usePortal: false, contain: false};
 
 export const KeyboardNavigationInsidePortalNoContain = Template().bind({});
 KeyboardNavigationInsidePortalNoContain.args = {usePortal: true, contain: false};
+
+export const RestoreFocusRef = Template().bind({});
+RestoreFocusRef.args = {usePortal: false, useRestoreFocusRef: true};
+
+export const RestoreFocusRefInsidePortal = Template().bind({});
+RestoreFocusRefInsidePortal.args = {usePortal: true, useRestoreFocusRef: true};
```

Comment:
Strange, this story restores focus back to the trigger button instead of the referenced input element but specifically for the dialogs beyond the first (aka when you open multiple dialogs). useRestoreFocus seems to call restoreElement on the external input element properly, but it seems like something else focuses the nested dialog trigger button afterwards...

Reply:
@LFDanLu The `onFocus` event handler on the document forces focus back to the `activeScope` at:
https://github.com/adobe/react-spectrum/blob/19ada7736dec4687b91d3ca758a0570e1215c857/packages/%40react-aria/focus/src/FocusScope.tsx#L275-L281. This kind of makes sense. If you have a FocusScope that contains focus open, like a modal dialog, you don't want a user be able to force focus to some element underneath it.

Reply:
I agree, I think that prioritizing a restoreElement within top most active FocusScope over the user provided restoreElement ref makes sense. The thing that throws me off is that this [story](https://reactspectrum.blob.core.windows.net/reactspectrum/19ada7736dec4687b91d3ca758a0570e1215c857/storybook/index.html?path=/story/focusscope--restore-focus-ref) **always** moves focus to the referenced input element.
Reproduction steps:
1. go to https://reactspectrum.blob.core.windows.net/reactspectrum/19ada7736dec4687b91d3ca758a0570e1215c857/storybook/index.html?path=/story/focusscope--restore-focus-ref
2. Open multiple dialogs by clicking on the "open dialog" button for each sub dialog that opens
3. Close the "top" most dialog (aka the last one you opened) via the "close" button
4. Note that focus is moved to the "Restore focus here" input element even though there are still other open dialogs.
5. Try steps 1-3 again, but instead close the dialog via the escape key. Note that focus is to the preceeding dialog's trigger button (this is what I would expect should happen)

Contrast this vs the [portal story](https://reactspectrum.blob.core.windows.net/reactspectrum/19ada7736dec4687b91d3ca758a0570e1215c857/storybook/index.html?path=/story/focusscope--restore-focus-ref-inside-portal) which always restores focus to a preceding dialog's trigger button if there is one available.

Reply:
Just wanna bring up https://github.com/adobe/react-spectrum/pull/2445/files#r828533708 again, perhaps we won't run into this weirdness if we strip out support for a user provided restoreElement ref...

Reply:
> The thing that throws me off is that this [story](https://reactspectrum.blob.core.windows.net/reactspectrum/19ada7736dec4687b91d3ca758a0570e1215c857/storybook/index.html?path=/story/focusscope--restore-focus-ref) **always** moves focus to the referenced input element.

Ok, I understand. It seems like we have a timing issue. With the portal we seem to execute the code I reference in https://github.com/adobe/react-spectrum/pull/2445#discussion_r833781088 after focus restores to the focus is restored to the restoreFocus ref.